src/runCompletion.ts

```diff
@@ -52,3 +53,22 @@ function getMaxResults(): number {
 
   return MAX_NUM_RESULTS;
 }
+
+type KnownLanguageType = keyof typeof languages;
+
+export function isKnownLanguage(languageId: string): languageId is KnownLanguageType {
+  return languageId in languages;
+}
+
+export function getFileNameWithExtension(
+  document: TextDocument
+): string {
+  const { languageId } = document;
+  let filenameWithExtension = document.fileName;
+  if (document.isUntitled) {
+    if (isKnownLanguage(languageId)) {
+      filenameWithExtension = document.fileName.concat(languages[languageId]);
+    }
```

Comment:
i'd create a function `getFileExtension(languageId: string): string | undefined` instead of `isKnownLanguage`, so that this code can look like so:

```typescript
const fileExtension = getFileExtension(languageId);
if (fileExtension) {
    filenameWithExtension = document.fileName.concat(fileExtension);
}
```